aemcmc/dists.py

```diff
@@ -50,30 +52,93 @@ def multivariate_normal_rue2005(rng, b, Q):
     return u + v
 
 
-# A and omega are assumed to be diagonal
-def multivariate_normal_cong2017(rng, A, omega, phi, t):
-    """
-    Sample from a multivariate normal distribution with a structured mean
-    and covariance matrix, as described in Example 4 [page 17] of [1]. This
-    algorithm is suitable for high-dimensional regression problems and the
-    runtime scales linearly with the number of regression coefficients.
+def multivariate_normal_cong2017(
+    rng: RandomStream,
+    A: TensorVariable,
+    omega: TensorVariable,
+    phi: TensorVariable,
+    t: TensorVariable,
+) -> TensorVariable:
+    r"""
+    Sample from a multivariate normal distribution with a structured mean and covariance.
+
+    As described in Example 4 [page 17] of [1], The covariance of this normal
+    distribution should be decomposable into a sum of a positive-definite matrix
+    and a low-rank symmetric matrix such that:
+
+    .. math::
+
+         \mathcal{N}(\mathbf{\Lambda}^{-1}\mathbf{\Phi}^T\mathbf{\Omega t}, \mathbf{\Lambda}^{-1})
+
+    where
+
+    .. math::
+
+        \begin{align*}
+            \mathbf{\Lambda} = (\mathbf{A}+\mathbf{\Phi}^T\mathbf{\Omega \Phi})
+        \end{align*}
 
-    This implementation assumes that A and omega are  diagonal matrices and
-    the parameters ``A`` and ``omega`` are expected to be an arrays containing
+    and :math:`\mathbf{A}` is the positive-definite part and
+    :math:`\mathbf{\Phi}^T\mathbf{\Omega \Phi}` is the eigen-factorization of
+    the low-rank part of the "structured" covariance.
+
+    Parameters
+    ----------
+    rng: TensorVariable
+        The random number generating object to be used during sampling.
+    A: TensorVariable
+        The entries of the diagonal elements of the positive-definite part of
+        the structured covariance.
+    omega: TensorVariable
+        The elements of the diagonal matrix in the eigen-decomposition of the
+        low-rank part of the structured covariancec of the multivariate normal
+        distribution.
+    phi: TensorVariable
+        A matrix containing the eigenvectors of the eigen-decomposition of the
+        low-rank part of the structured covariance of the normal distribution.
+    t: TensorVariable
+        A 1D array whose length is the number of eigenvalues of the low-rank
+        part of the structured covariance.
+
+    Notes
+    -----
+    This algorithm is suitable for high-dimensional regression problems and the
+    runtime scales linearly with the number of regression coefficients. This
+    implementation assumes that `A` and `omega` are diagonal matrices and
+    the parameters `A` and ``omega`` are expected to be 1-dimension containing
```

Comment:
"are expected to be vectors that contain the diagonal entries of the respective matrices they represent"